staging/src/k8s.io/client-go/transport/cache.go

```diff
@@ -138,6 +147,21 @@ func (c *tlsTransportCache) get(config *Config) (http.RoundTripper, error) {
 	return transport, nil
 }
 
+type upgradeAwareTransport struct {
+	h1, h2 http.RoundTripper
+}
+
+func (tr *upgradeAwareTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	if v := req.Header.Get("Upgrade"); v != "" {
```

Comment:
I think it would be more correct to check the map for the presence of the key instead of silently skipping this `if` statement if the upgrade header is set to the empty string.

Reply:
`httpstream.IsUpgradeRequest` might be appropriate?

Reply:
PTAL at new upgrade check.